### PR TITLE
Ignore account hints when prompt=select_account

### PR DESCRIPTION
**Pull request by tnorling**
AAD throws an error when you pass both `prompt=select_account` and a `login_hint` or `sid` as these are conflicting. This becomes problematic when someone wants to use `prompt=select_account` to switch users and has previously used the `setActiveAccount` API in browser (we quietly pass down the active account if none is provided to the interactive API).

This PR ignores account hints if `prompt=select_account` to prevent this error from happening unexpectedly.

### CODE REVIEW

File: lib/msal-common/src/client/AuthorizationCodeClient.ts [hunk 1]
```diff
@@ -248,21 +248,26 @@ export class AuthorizationCodeClient extends BaseClient {
...
 
         // Add sid or loginHint with preference for sid -> loginHint -> username of AccountInfo object
-        if (request.sid) {
-            parameterBuilder.addSid(request.sid);
-        } else if (request.account) {
-            const accountSid = this.extractAccountSid(request.account);
-            // If account and loginHint are provided, we will check account first for sid before adding loginHint
-            if (accountSid) {
-                parameterBuilder.addSid(accountSid);
+        if (request.prompt !== PromptValue.SELECT_ACCOUNT) {
```
Comment (jasonnutter) [inline lib/msal-common/src/client/AuthorizationCodeClient.ts:251]: Would we want to also ignore account hints for login?

**Review by jasonnutter — APPROVED** [review]
One question otherwise looks good!

File: lib/msal-common/src/client/AuthorizationCodeClient.ts (hunk 1 above)
Comment (tnorling) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:251]: I went back and forth on that. In the `prompt=login` scenario `login_hint` could be used to pre-fill the email field so maybe just ignore `sid`?

Comment (jasonnutter) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:251]: I think I've seen scenarios where an app would want to trigger the login prompt with the current user, so as long as we provide a way to set prompt=login with and without an account hint, we should be good.

Comment (jasonnutter) [reply lib/msal-common/src/client/AuthorizationCodeClient.ts:251]: And yeah, sid can be ignored for interactive requests.

**Comment by sameerag** [comment]
The server mandates that sid/login_hint should be used only for `prompt=none` use cases. Did we remove that check?

**Comment by sameerag** [comment]
The server mandates that sid/login_hint should be used only for `prompt=none` use cases. Did we remove that check?

**Comment by tnorling** [comment]
> The server mandates that sid/login_hint should be used only for `prompt=none` use cases. Did we remove that check?

AFAIK this was never present in 2.x. This PR only sends `sid` when `prompt=none`, however, `login_hint` is valid for `prompt=login` so will only be ignored if `prompt=select_account`.

**Review by jasonnutter — APPROVED** [review]

**Review by sameerag — APPROVED** [review]
Approved. One thing: test cases to ignore sid for prompt!=none (if user provides it), please check that once.

**Comment by tnorling** [comment]
> Approved. One thing: test cases to ignore sid for prompt!=none (if user provides it), please check that once.

Yes, it's there (test cases to use login_hint over sid if prompt != none covers this as login_hint has lower priority than sid)

**Comment by sameerag** [comment]
> Yes, it's there (test cases to use login_hint over sid if prompt != none covers this as login_hint has lower priority than sid)

I don't think we are testing this use case: Only `sid` and ignoring it if passed with `prompt != none`. This is different from making `sid` lower priority. Not blocking this PR, but this test case is critical to have around, in case something changes tomorrow.

**Comment by tnorling** [comment]
> > Yes, it's there (test cases to use login_hint over sid if prompt != none covers this as login_hint has lower priority than sid)
> 
> I don't think we are testing this use case: Only `sid` and ignoring it if passed with `prompt != none`. This is different from making `sid` lower priority. Not blocking this PR, but this test case is critical to have around, in case something changes tomorrow.

Gotcha, added!

**Pull request merged**